src/utransport.rs

```diff
@@ -25,6 +25,32 @@ use zenoh::{
     queryable::Query,
 };
 
+macro_rules! error_callback {
```

Comment:
Cool uses of `macro_rules!` :slightly_smiling_face:

Reply:
After thinking twice, I found that maybe I can use inline now since there is no type-related issue anymore.